smart_open/smart_open_lib.py

```diff
@@ -754,11 +754,11 @@ def HttpOpenRead(parsed_uri, mode='r', **kwargs):
 
     response = HttpReadStream(url, **kwargs)
 
-    fname = url.split('/')[-1]
+    fname = urlsplit(url, allow_fragments=False).path.split('/')[-1]
```

Comment:
Why you use `allow_fragments=False` ? I get the same result with/without this flag

```python
In [13]: urlsplit('http://127.0.0.1/data.gz', allow_fragments=False)
Out[13]: SplitResult(scheme='http', netloc='127.0.0.1', path='/data.gz', query='', fragment='')

In [14]: urlsplit('http://127.0.0.1/data.gz', allow_fragments=True)
Out[14]: SplitResult(scheme='http', netloc='127.0.0.1', path='/data.gz', query='', fragment='')

In [15]: urlsplit('http://127.0.0.1/data.gz?a=3,b=4', allow_fragments=False)
Out[15]: SplitResult(scheme='http', netloc='127.0.0.1', path='/data.gz', query='a=3,b=4', fragment='')

In [16]: urlsplit('http://127.0.0.1/data.gz?a=3,b=4', allow_fragments=True)
Out[16]: SplitResult(scheme='http', netloc='127.0.0.1', path='/data.gz', query='a=3,b=4', fragment='')
```